test/unit/utils/JobRoleUtilTests.ts

```diff
@@ -11,3 +12,67 @@ describe('Format Date to LocalDateString', () => {
     assert.equal(resultDate, 'N/A');
   });
 });
+
+describe('extractJobRoleFilterParams', () => {
+  it('should correctly extract filter parameters from request query', () => {
+    const mockRequest = {
+      query: {
+        roleName: 'Developer',
+        jobRoleLocation: 'NYC',
+        capabilityId: '1',
+        bandId: '2',
+        closingDate: '2023-09-01',
+      },
+    }as unknown as express.Request;
```

Comment:
do we need both as unknown ad express.Request? Would not as express.Request be enough?

Reply:
From what I checked, both elements are needed for proper operation

![Screenshot 2024-09-04 at 10 27 54](https://github.com/user-attachments/assets/9ce80d71-f4f1-4975-9e2e-34cf60971fb8)